php-checks/src/main/java/org/sonar/php/checks/ClassNameCheck.java

```diff
@@ -55,15 +56,18 @@
   @Override
   public void init() {
     pattern = Pattern.compile(format);
-    subscribeTo(PHPGrammar.CLASS_DECLARATION);
   }
 
   @Override
-  public void visitNode(AstNode astNode) {
-    String className = astNode.getFirstChild(PHPGrammar.IDENTIFIER).getTokenOriginalValue();
+  public void visitClassDeclaration(ClassDeclarationTree tree) {
+    super.visitClassDeclaration(tree);
+
+    String className = tree.name().text();
 
     if (!pattern.matcher(className).matches()) {
-      getContext().createLineViolation(this, "Rename class \"{0}\" to match the regular expression {1}.", astNode, className, format);
+      String message = String.format(MESSAGE, className, this.format);
+      context().newIssue(this.KEY, message)
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Change this instance-reference to a static reference. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS2209)